contributors/design-proposals/scheduling/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,186 @@
+type ResourceQuotaSpec struct {
+        ...
+
+        // A collection of filters that must match each object tracked by a quota.
+        // If not specified, the quota matches all objects.
+        // +optional
+        Scopes []ResourceQuotaScope
+}
+```
+Four new `ResourceQuotaScope` will be defined for matching pods based on priority class names.
+```go
+// A ResourceQuotaScope defines a filter that must match each object tracked by a quota
+type ResourceQuotaScope string
+
+const (
+        ...
+        ResourceQuotaScopePriorityClassNameExists ResourceQuotaScope = "PriorityClassNameExists"
+        // Match all pod objects that do not have any priority class mentioned
+        ResourceQuotaScopePriorityClassNameNotExists ResourceQuotaScope = "PriorityClassNameNotExists"
+        // Match all pod objects that have priority class from the set
+        ResourceQuotaScopePriorityClassNameIn ResourceQuotaScope = "PriorityClassNameIn"
+        // Match all pod objects that do not have priority class from the set
+        ResourceQuotaScopePriorityClassNameNotIn ResourceQuotaScope = "PriorityClassNameNotIn"
+)
+```
+
+### Changes in Admission Controller Configuration
+
+A new field `MatchScopes` will be added to `Configuration.LimitedResource`. `MatchScopes` will be a collection of one or more of the four newly added priority class based `Scopes` that are explained in above section.
+
+```go
+// Configuration provides configuration for the ResourceQuota admission controller.
+type Configuration struct {
+        ...
+        LimitedResources []LimitedResource
+}
+
+// LimitedResource matches a resource whose consumption is limited by default.
+// To consume the resource, there must exist an associated quota that limits
+// its consumption.
+type LimitedResource struct {
+        ...
+
+        // MatchScopes is a collection of filters based on priority classes.
+        // If the object in the intercepted request matches these rules,
+        // quota system will ensure that corresponding quota MUST have
+        // priority based Scopes matching the object in request. 
+        //
+        // If MatchScopes has matched on an object, request for the resource will be denied 
+        // if there is no quota with matching Scopes. In this case, matching priority class based Scopes
+        // will be an additional requirement for any quota to qualified as covering quota.
+        // +optional
+        MatchScopes []string `json:"matchScopes,omitempty"`
+}
+```
+
+### Expected Behavior of ResourceQuota Admission Controller and ResourceQuota System
+`MatchScopes` will be configured in admission controller configuration to apply quota based on priority class names. If `MatchScopes` matches/selects an incoming pod request, request will be **denied if a Covering Quota is missing**. The meaning of Covering Quota is: any quota which has priority class based `Scopes` that matches/selects the pod in the request. 
```

Comment:
nit: its really any matching scope, but in this case we are using it for priority.  any documentation that is written should keep this generic and maybe describe some concrete examples.